transports/transports-deciding.md

```diff
@@ -0,0 +1,196 @@
+
+#### Disadvantages
+
+- Does not offer a native publish-subscribe mechanism, therefore it requires NServiceBus persistence to be configured for storing event subscriptions. [Explicit routing for publish/subscribe](/nservicebus/messaging/routing.md#event-routing-message-driven) must also be specified.
+- Many organizations don't have the same level of operational expertise with MSMQ that they do with other technologies (e.g. SQL Server), so it may require additional training.
+- Scaling out compute-intensive workload require setting up the distributor or sender-side distribution of messages to destination queues. This is harder than with the competing consumer pattern that other transports support.
+
+#### When would you select this transport
+
+- If you need store & forward for a higher reliability that your queuing technology is always available.
+- If you’re running a Windows environment on-premise and don’t want to spend additionally on licenses and/or training.
+- If you are able to coop distributed transactions and want the reliability it adds.
+
+
+
+## Azure Service Bus Transport
+
+Azure provides multiple messaging technologies and one of the most advanced and reliable ones is Azure Service Bus.
+
+Advantages
+
+- No need to maintain infrastructure or have deep knowledge of Azure Service Bus as Microsoft and NServiceBus take care of almost everything.
+- Ability to scale in both features and price range.
+- AMQP based implementation
+- Supports messaging transactions unlike other messaging services on Azure
+- Up to 1MB message size
+- Levering native capabilities of the transport instead of taxing an endpoint
+
+## Disadvantages
+
+- Requires monitoring of costs, although NServiceBus supports tweaking number of transactions to Azure.
+- No on-premise option for development or testing
+- Processing a message has to complete within 5 minutes
+- Depends on TCP and might require opening additional ports in the firewall
+
+#### When would you select this transport
+
+- If your application is running on Windows Azure
+- If you want enterprise messaging features, like additional reliability
+- If you want a bigger message size than Azure Storage Queues
+- If you want to scale up and out with your business needs
+
+
+
+## Azure Storage Queues
+
+Azure Storage Queues is a service hosted on the Azure platform. Compared to Azure Service Bus it has less advanced features but is also more cost effective.
+
+#### Advantages
+
+- Can store very large numbers of messages (up to 200 TB limit of the Azure Storage account). Even though you should not store that many messages, it is a feature we wanted to add.
+- Very low price per message
+- Very high availability
+
+#### Disadvantages
+
+- The size of a single message is limited to 64 KB including headers and body. NServiceBus adds a stack-trace to the message before sending it to the error queue.
+- Atomic message receive and data manipulation requires Outbox
```

Comment:
Feels like there's a mix of "Sends atomic with Receive" guarantee and "message and business data processing".

Reply:
While you can technically use SQLP/NH against SqlAzure do we have any concerns mentioning the outbox since ASQ's "low cost, high throughput, high availability" benefits seem to not match that well with SqlAzure usage?

Ie: Is that fact that we don't have a "cloud native" outbox an issue?

Reply:
The line has been removed.